pkg/sdn/plugin/subnets.go

```diff
@@ -75,26 +79,29 @@ func (master *OsdnMaster) addNode(nodeName string, nodeIP string, hsAnnotations
 		}
 	}
 
-	// Create new subnet
-	sn, err := master.subnetAllocator.GetNetwork()
-	if err != nil {
-		return "", fmt.Errorf("error allocating network for node %s: %v", nodeName, err)
-	}
-
-	sub = &osapi.HostSubnet{
-		TypeMeta:   metav1.TypeMeta{Kind: "HostSubnet"},
-		ObjectMeta: metav1.ObjectMeta{Name: nodeName, Annotations: hsAnnotations},
-		Host:       nodeName,
-		HostIP:     nodeIP,
-		Subnet:     sn.String(),
-	}
-	sub, err = master.osClient.HostSubnets().Create(sub)
-	if err != nil {
-		master.subnetAllocator.ReleaseNetwork(sn)
-		return "", fmt.Errorf("error creating subnet %s for node %s: %v", sn.String(), nodeName, err)
+	// Create new subet
+	for _, possibleSubnet := range master.subnetAllocatorList {
+		sn, err := possibleSubnet.GetNetwork()
+		if err == netutils.ErrSubnetAllocatorFull {
+			log.Infof("error allocating network for node %s:%v in subnet %s", nodeName, err)
```

Comment:
It is not an error when one of the subnet is fully consumed, we just need to assign from rest of the available subnets.
May be:
``` 
if err != nil {
    if err != netutils.ErrSubnetAllocatorFull {
        log.Errorf("Error fetching network ...")
    }
}
// Got valid sn, process further
```